vertx-web/src/main/java/io/vertx/ext/web/handler/impl/StaticHandlerImpl.java

```diff
@@ -258,54 +253,41 @@ private void sendDirectory(RoutingContext context, String path, String file) {
       if (classLoader == null) {
         return callable.call();
       } else {
-        final ClassLoader original = Thread.currentThread().getContextClassLoader();
-        try {
-          Thread.currentThread().setContextClassLoader(classLoader);
-          return callable.call();
-        } finally {
-          Thread.currentThread().setContextClassLoader(original);
+        synchronized (this) {
```

Comment:
I don't understand why you need a synchronized block here. You are changing the current thread classloader, there shouldn't be any concurrency issues.

Or do I miss something?

Reply:
the old code was synchronizing on the methods that would call this. Let me check all the paths that lead to there so see if i can safely remove it.

Reply:
I don't believe it is required here. Thread accesses are fine (by definition).

Reply:
It was been removed from 3.8 (however this doesn't affect us as it was a OSGi thing) quarkus and no vertx app i could grep in github makes use of this classloader.